osu.Game/Overlays/Mods/BpmDisplay.cs

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Utils;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Mods;
+using osuTK;
+
+namespace osu.Game.Overlays.Mods
+{
+    public sealed class BpmDisplay : CompositeDrawable
+    {
+        [Resolved]
+        private OsuColour colours { get; set; } = null!;
+
+        [Resolved]
+        private OverlayColourProvider colourProvider { get; set; } = null!;
+
+        [Resolved]
+        private IBindable<IReadOnlyList<Mod>> mods { get; set; } = null!;
+
+        [Resolved]
+        private IBindable<WorkingBeatmap> working { get; set; } = null!;
+
+        private readonly BindableNumber<int> current = new BindableNumber<int>();
+
+        private ModSettingChangeTracker? settingChangeTracker;
+        private const float bpm_value_area_width = 56;
+        private const float transition_duration = 200;
+        private readonly Box underlayBackground;
+        private readonly Box contentBackground;
+        private readonly BpmCounter counter;
+        private readonly OsuSpriteText dashText;
+
+        public BpmDisplay()
+        {
+            Height = DifficultyMultiplierDisplay.HEIGHT;
+            AutoSizeAxes = Axes.X;
+
+            InternalChild = new InputBlockingContainer
```

Comment:
Having mixed feelings about all of this layout setup being essentially copy-pasted from `DifficultyMultiplierDisplay`. Is this really necessary?